### PR TITLE
feat(category): add ability to process range type filters

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

Note: The `DaffCategoryFromToFilterSeparator` is not needed yet, but will be needed if a ranged type filter value from a platform ever differs. For example, price: "30_40" vs price: "30-40"

### CODE REVIEW

File: libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts [hunk 1]
```diff
@@ -16,9 +16,18 @@ export class DaffMagentoAppliedFiltersTransformService {
...
-			magentoFilters[filter.name] = {
-				...magentoFilters[filter.name],
-				[this.transformActionEnum(filter.action)]: filter.value
+			if(filter.action === DaffCategoryFilterActionEnum.FromTo) {
+				const fromToValues = filter.value.split('-');
+				magentoFilters[filter.name] = {
+					...magentoFilters[filter.name],
+					[MagentoCategoryFilterActionEnum.From]: fromToValues[0],
+					[MagentoCategoryFilterActionEnum.To]: fromToValues[1]
+				}
```
Comment (griest024) [inline libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts]: I think you should add a brief comment here explaining why `DaffCategoryFilterActionEnum.FromTo` needs such special treatment.

File: libs/category/src/drivers/magento/transformers/category-page-config-transformer.service.ts [hunk 2]
```diff
@@ -40,6 +40,19 @@ export class DaffMagentoCategoryPageConfigTransformerService {
     }
 	}
 
+	// There is no way to determine what type these filters are except through the
+	// Magento documentation.
+	private transformAggregateType(attribute_code: string): DaffCategoryFilterType {
```
Comment (griest024) [inline libs/category/src/drivers/magento/transformers/category-page-config-transformer.service.ts]: I think you should type `attribute_code` as `MagentoAggregation['attribute_code']`.

File: libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts [hunk 3]
```diff
@@ -16,9 +16,18 @@ export class DaffMagentoAppliedFiltersTransformService {
 		};
 
 		daffFilters.forEach(filter => {
-			magentoFilters[filter.name] = {
-				...magentoFilters[filter.name],
-				[this.transformActionEnum(filter.action)]: filter.value
+			if(filter.action === DaffCategoryFilterActionEnum.FromTo) {
+				const fromToValues = filter.value.split('-');
```
Comment (griest024) [inline libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts]: Should this be referencing `DaffCategoryFromToFilterSeparator` for the `split` argument?

File: libs/category/src/drivers/magento/transformers/category-page-config-transformer.service.ts (hunk 2 above)
Comment (lderrickable) [reply libs/category/src/drivers/magento/transformers/category-page-config-transformer.service.ts]: done

File: libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts (hunk 3 above)
Comment (lderrickable) [reply libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts]: yea, definitely. Changed

File: libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts (hunk 1 above)
Comment (lderrickable) [reply libs/category/src/drivers/magento/transformers/applied-filter-transformer.service.ts]: done

**Pull request merged**